pkg/descheduler/strategies/pod_lifetime.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategies
+
+import (
+	v1 "k8s.io/api/core/v1"
+	v1meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"sigs.k8s.io/descheduler/pkg/api"
+	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+)
+
+// PodLifeTime evicts pods on nodes that were created more than strategy.Params.MaxPodLifeTimeSeconds seconds ago.
+func PodLifeTime(client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, evictLocalStoragePods bool, podEvictor *evictions.PodEvictor) {
+	for _, node := range nodes {
+		klog.V(1).Infof("Processing node: %#v", node.Name)
+		pods := listOldPodsOnNode(client, node, strategy.Params.MaxPodLifeTimeSeconds, evictLocalStoragePods)
+		for _, pod := range pods {
+			success, err := podEvictor.EvictPod(pod, node)
```

Comment:
If possible, we should check if any evicted pod is still schedulable to at least one node. In this case it can be the same.

Reply:
@ingvagabund I want to make sure I understand your suggestion correctly. You think the code should check if the pod can be rescheduled on at least one other node before evicting it?

Something like ...
```go
// podCanBeRescheduled is a bool that determines if the pod that will be evicted can be scheduled on at least one other node
if podCanBeRescheduled {
    success, err := podEvictor.EvictPod(pod, node)
} else {
    klog.V(1).Infof("Skipping pod: %#v\n because it cannot be rescheduled", pod.Name)
}
```

Reply:
> Something like ...
yes,

e.g. if there's still a node whose taints are respected. Some function/method that will be called across all strategies. Though, not a blocker for the PR.